packages/evolution-frontend/src/components/inputs/InputString.tsx

```diff
@@ -53,9 +53,8 @@ export const InputString = <CustomSurvey, CustomHousehold, CustomHome, CustomPer
             id={props.id}
             value={value}
             onBlur={props.onValueChange}
-            // if disallowedCharactersRegex is null or undefined, then replace(...) does nothing, which is the intended behavior.
             onChange={(e) =>
-                setValue(e.target.value.replace(props.widgetConfig.disallowedCharactersRegex as RegExp, ''))
+                setValue(!_isBlank(props.widgetConfig.inputFilter) ? props.widgetConfig.inputFilter!(e.target.value) : e.target.value)
```

Comment:
C'est quoi de ! après inputFilter?

Reply:
@tahini 
C'est le "non null assertion operator", afin d'éviter l'erreur `Object is possibly 'undefined'. ts(2532)`.
Avec `_isBlank` on vérifie que `inputFilter !== undefined`, donc on peut utiliser l'opérateur `!`


https://blog.logrocket.com/understanding-exclamation-mark-typescript/#what-exclamation-mark-does-typescript

Reply:
Nice!